aggregator/apps/api/v1/stitcher.py

```diff
@@ -2,6 +2,43 @@
 from django.urls import reverse
 
 
+def ballot_charisma(ballot, sort_keys):
+    charisma_map = {
+        "ref": {"default": 100},
+        "parl": {"default": 90},
+        "europarl": {"default": 80},
+        "mayor": {"default": 70, "local-authority": 65},
+        "naw": {"default": 60},
+        "sp": {"default": 60},
+        "nia": {"default": 60},
+        "gla": {"default": 60},
+        "pcc": {"default": 50},
+        "local": {"default": 40},
+    }
+    modifier = 0
+    ballot_paper_id = ballot["ballot_paper_id"]
+    values_for_type = charisma_map[ballot_paper_id.split(".")[0]]
+    base_charisma = values_for_type.get(
+        sort_keys.get(ballot_paper_id), values_for_type.get("default")
+    )
+    # GLA Additional are less important than constituencies
+    if ballot_paper_id.startswith("gla.a"):
+        modifier += 1
```

Comment:
Rather than hard-coding this one case, do you think it would make more sense to do subtypes as a sort key and define something like this:

```
"gla": {"default": 60, "c": 60, "a": "55"},
"naw": {"default": 60, "c": 60, "r": "55"},
"sp": {"default": 60, "c": 60, "r": "55"},
```

Reply:
Good idea. I've left out the `c` part, as that's the default